File: CrittersWeb/DtoModels/Game/HeroDto.cs
using CrittersWeb.Critters.Modeles;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace CrittersWeb.DtoModels.Game
{
    public class HeroDto
    {
        public int Id { get; set; }
        public string GameUserId { get; set; }        
        public int X { get; set; }
        public int Y { get; set; }
        public Direction Direction { get; set; }
        public int Energy { get; set; }
        public int CellsInBuffer { get; set; }
        public int CellsBufferSize { get; set; }
        public int MaxEnergy { get; set; }
        public int FieldSize { get; set; }
        public int CellsLimitToPut { get; set; }

    }
}

File: CrittersWeb/client/src/app/dto/game/HeroDto.ts
export class HeroDto {
    id!: number;
    gameUserId!: string;
    x!: number;
    y!: number;
    direction!: Direction;
    energy!: number;
    cellsInBuffer!: number;
    cellsBufferSize!: number;
    maxEnergy!: number;
    fieldSize!: number;
    cellsLimitToPut!: number;
}

export enum Direction {
    top = 0,
    right = 1,
    bottom = 2,
    left = 3
}